astropy/coordinates/earth.py

```diff
@@ -619,18 +644,42 @@ def get_gcrs_posvel(self, obstime):
         obsgeovel : `~astropy.coordinates.CartesianRepresentation`
             The GCRS velocity of the object
         """
-        # do this here to prevent a series of complicated circular imports
-        from .builtin_frames import GCRS
-
-        itrs = self.get_itrs(obstime)
-        geocentric_frame = GCRS(obstime=obstime)
         # GCRS position
-        obsgeoloc = itrs.transform_to(geocentric_frame).cartesian
-        vel_x = -OMEGA_EARTH * obsgeoloc.y
-        vel_y = OMEGA_EARTH * obsgeoloc.x
-        vel_z = 0. * vel_x.unit
-        obsgeovel = CartesianRepresentation(vel_x, vel_y, vel_z)
-        return obsgeoloc, obsgeovel
+        gcrs_data = self.get_gcrs(obstime).data
+        obsgeopos = gcrs_data.without_differentials()
+        obsgeovel = gcrs_data.differentials['s'].to_cartesian()
+        return obsgeopos, obsgeovel
+
+    def gravitational_redshift(self, obstime):
```

Comment:
similar here with the above re: privitizing this for this PR.  One could *definitely* see people wanting to use this on its own, but we should then have it in the changelog.

Reply:
👍